src/Core/Material/BlinnPhongMaterialModel.cpp

```diff
@@ -24,6 +26,81 @@ void BlinnPhongMaterialModel::displayInfo() const {
     print( hasNormalTexture(), " Normal Texture : ", m_texNormal );
     print( hasOpacityTexture(), " Alpha Texture  : ", m_texOpacity );
 }
+Utils::Color Material::BlinnPhongMaterialModel::evalBSDF( Vector3f w_i,
+                                                          Vector3f w_o,
+                                                          Vector3f normal,
+                                                          Vector2f uv ) {
+    // diffuse lambertien component
+    Utils::Color diffuse = m_kd / M_PI;
+
+    // Blinn-Phong specular component
+    Vector3f halfway = w_i + w_o;
+    halfway.normalize();
+    Scalar specularIntensity = (m_ns + 2.0f) / (2.0f * M_PI) * std::pow(normal.dot(halfway), m_ns);
+    Utils::Color specular = m_ks * specularIntensity;
+
+    // Combine the diffuse and specular components
+    Utils::Color bsdf = diffuse + specular;
+}
+
+std::optional<std::pair<Vector3f, Scalar>>
+BlinnPhongMaterialModel::sample( Vector3f inDir, Vector3f normal, Vector2f u ) {
+    /*
+    compute intensity of diffuse and specular
+    compute norm (max(1, Idiffuse + Ispecular))
+    normalize Idiffuse and Ispecular
+    */
+
+    Vector3f nextDir, halfway;
+
+    Scalar dIntensity = (m_kd.x() + m_kd.y() + m_kd.z()) / 3.f;
```

Comment:
Assuming the `m_kd` color is in linear RGB color space, that is the color space used in Radium, the luminance (what you call intensity here) should be computed according to the following : 
```suggestion
    Vector3  rgbToLuminance { 0.2126_ra , 0.7152_ra,  0.0722_ra };
    Scalar dIntensity = m_kd.dot( rgbToLuminance );
```